engine/access/rest/websockets/handler_test.go

```diff
@@ -0,0 +1,91 @@
+package websockets_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/access/rest/websockets"
+	"github.com/onflow/flow-go/engine/access/state_stream/backend"
+	streammock "github.com/onflow/flow-go/engine/access/state_stream/mock"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+var (
+	chainID = flow.Testnet
+)
+
+type WsHandlerSuite struct {
+	suite.Suite
+
+	logger       zerolog.Logger
+	handler      *websockets.Handler
+	wsConfig     *websockets.Config
+	streamApi    *streammock.API
+	streamConfig *backend.Config
+}
+
+func (s *WsHandlerSuite) SetupTest() {
+	s.logger = unittest.Logger()
+	s.wsConfig = websockets.NewDefaultWebsocketConfig()
+	s.streamApi = streammock.NewAPI(s.T())
+	s.streamConfig = &backend.Config{}
+	s.handler = websockets.NewWebSocketHandler(s.logger, s.wsConfig, chainID.Chain(), s.streamApi, *s.streamConfig, 1024)
+}
+
+func TestWsHandlerSuite(t *testing.T) {
+	suite.Run(t, new(WsHandlerSuite))
+}
+
+func ClientConnection(url string) (*websocket.Conn, *http.Response, error) {
+	wsURL := "ws" + strings.TrimPrefix(url, "http")
+	return websocket.DefaultDialer.Dial(wsURL, nil)
+}
+
+func (s *WsHandlerSuite) TestSubscribeRequest() {
```

Comment:
This test is only to make sure we get back data from data providers. this should removed in favor of https://github.com/onflow/flow-go/issues/6631